compressed_string.py

```diff
@@ -0,0 +1,36 @@
+def compress_string(string: str) -> str:
+    """
+    Function which takes a string and then compresses it.
+    """
+    
+    # intialise vairables
+    count = 1
+    compressed_string = ""
+    
+    letters = [char for char in string if char.isalpha()]  # remove all digits from the string
+    letters = "".join(letters)
+
+    for char in range(len(letters)):
+
+        if char + 1 > len(letters) - 1:  # check to make sure that the range is not out of index
+            if count != 1:
+                compressed_string = compressed_string + letters[char] + str(count)
+            else:
+                compressed_string += letters[char]
+            break
```

Comment:
Be careful with nested `if` statements too. They are very delicate and if you don't stop it early, it's so easy to have very long files with all kinds of if/else logic 🥲